### PR TITLE
Add device service raml (#352)

### PR DESCRIPTION
**Pull request by iain-anderson**
Signed-off-by: Iain Anderson <iain@iotechsys.com>

### CODE REVIEW

File: api/raml/device-sdk.raml [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    -
+        reading: '{"type":"object","$schema":"http://json-schema.org/draft-06/schema#","title":"reading","properties":{"name":{"type":"string"},"value":{"type":"string"}}}'
+    -
+        setting: '{"type":"object","$schema":"http://json-schema.org/draft-06/schema#","title":"setting","additionalProperties":{"type":"string"}}'
+    -
+        callbackalert: '{"type": "object","$schema": "http://json-schema.org/draft-06/schema#","title": "Notification Schema","properties": {"id": {"description": "the identifier of the object which is called back","type": "string"},"actionType": {"description": "the type of the called back object","enum":["PROFILE","DEVICE","PROVISIONWATCHER","SCHEDULE","SCHEDULEEVENT"],"type": "string"}},"required": ["id"]}'
+    -
+        versionobject: '{"type":"object","$schema":"http://json-schema.org/draft-06/schema#","title":"version","properties":{"version":{"type":"string"},"sdk_version":{"type":"string"}}}
```
Comment (cherrycl) [inline api/raml/device-sdk.raml]: Missing a single quote on the end of the line.

Comment (iain-anderson) [reply api/raml/device-sdk.raml]: updated

File: api/raml/device-sdk.raml [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    description: Example -- http://localhost:49999/api/v1/ping
+    get:
+        description: Test service providing an indication that the service is available.
+        displayName: service up check
+        responses:
+            "200":
+                body:
+                    text/plain:
+                        description: device service version string
+                        example: 'V1.0'
```
Comment (cloudxxx8) [inline api/raml/device-sdk.raml]: in device-sdk-go, it only returns 1.0.0 without "V".
does C SDK returns the version number with "V"?

File: api/raml/device-sdk.raml [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+                schema: setting
+                example: '{"AHU-TargetTemperature": "28.5"}'
+        responses:
+            "200":
+                description: The PUT commands were successful.
+            "423":
+                description: If the device service is locked (admin state).
+            "500":
+                description: The device driver is unable to process the request.
+    post:
```
Comment (cloudxxx8) [inline api/raml/device-sdk.raml]: We only support Put for command, so the Post command should be removed

**Comment by iain-anderson** [comment]
Closing. The doc requirement will be satisfied by adding an API definition in Swagger format instead.

**Pull request closed without merging**